schemas/styles/csl-names.rnc

```diff
@@ -0,0 +1,352 @@
+      affixes,
+      font-formatting,
+      text-case
+    }
+  name.name-type =
+    
+    ## Use to apply separate formatting to personal versus institutional names.
+    element cs:name-type {
+      attribute name { "personal" | "institutional" },
+      name.forms,
+      affixes,
+      font-formatting,
+      text-case,
+      name.name-part*
+    }
+  names.et-al =
+    
+    ## Specify the term used for et-al abbreviation and its formatting.
+    element cs:et-al {
+      
+      ## Select the term to use for et-al abbreviation.
+      [ a:defaultValue = "et-al" ]
+      attribute term { "et-al" | "and others" }?,
+      
+      ## The form of the cs:et-al term to use when et-al abbreviation is activated. 
+      ## Set to "none" to omit the term. Set to "long", "short", or "symbol" to
+      ## render the respective form of the term.
+      [ a:defaultValue = "long" ]
+      attribute form { "none" | "long" | "short" | "symbol" }?,
+      font-formatting
+    }
+  
+  ## If name.form="long", render the "institution-short" name-part.
+  names.abbreviation =
+    element cs:abbreviation { affixes, font-formatting, text-case }
+  
+  ## Render the "alternate" name and specify formatting.
+  names.alternate = element cs:alternate { names.name }
+  
+  ## Inherits variable from the parent cs:names element.
+  names.label =
+    element cs:label {
+      [ a:defaultValue = "long" ] attribute form { term.form }?,
+      
+      ## When multiple='combined', set the delimiter for collapsed name labels
+      ## (e.g., ", " in "Creator, Writer, & Director")
+      [ a:defaultValue = ", " ] delimiter,
+      
+      ## Specify when the label delimiter is used between the second-to-last and last label of a label list;
+      ## only has an effect when the "and" term is used (e.g., "Writer & Director" or "Writer, & Director").
+      [ a:defaultValue = "contextual" ]
+      attribute delimiter-precedes-last {
+        
+        ## The label delimiter is only used when the label list consists of
+        ## three or more labels.
+        "contextual"
+        | 
+          ## The label delimiter is always used.
+          "always"
+        | 
+          ## The label delimiter is never used.
+          "never"
+      }?,
+      
+      ## Use to separate the second-to-last and last label of a label list by the "and" term.
+      attribute and {
+        
+        ## Use the "and" term "long" form (e.g., "Writer and Director").
+        "long"
+        | 
+          ## Use the "and" term "symbol" form (e.g., "Writer & Director").
+          "symbol"
+        | 
+          ## Use the "and" term "short" form. Not common in English, but is seen in other languages.
+          ## In German, for example: "Drehbuchautor u. Regisseur, where "u." is short for "und".
+          "short"
+      }?
+    }
+  
+  ## Render the "place" associated with a name and specify formatting;
+  ## used to render a publisher's place.
+  names.place =
+    element cs:place {
```

Comment:
I don't see why this should be a dedicated element, when everywhere (or almost everywhere?) else in the schema variables are accessed with attributes. 

Can you clarify?

Reply:
It's like `label`, `abbreviation`, or `alternate`--this element controls the position, formatting, and affixes on the place relative to the `name`. This is consistent with how `names` formatting has always worked.

Reply:
How the affixes? I don't see them.

Reply:
By the way, I also don't see them on `cs:label`. What am I missing?

Reply:
> It's like `label`, `abbreviation`, or `alternate`--this element controls the position, formatting, and affixes on the place relative to the `name`. This is consistent with how `names` formatting has always worked.

I haven't been following the name stuff closely, but I'm not sure I buy this ATM.

Are we saying whenever we add a new variable to names we need a new CSL element, whereas everywhere else in the schema we'll use an attribute value?

Reply:
You're just not appreciating how the `names` element works (and has always worked). `names` defines a layout for name variables. The order, formatting, and spacing of its components have always been controlled by sub-elements of `names`. This is great because there is huge diversity in the placement, spacing, and formatting of name parts, labels, et-al, etc. I assume this is why you designed it this way in the first place. 

Placement of various pieces within `names` has never been done using attributes.

Previously, the sub-components of names were label, name, and et-al. We've added abbreviation and alternate. This is adding place. place is just like label in that styles vary in whether it is included, its position, its spacing and punctuation, its text formatting, etc. Adding place as an element within is entirely consistent with the existing design of `names`. 

To make place now be an attribute would be to make it. **unlike** every other part of name formatting. It would require a slew of new attributes: place-position, place-name-delimiter, place-text-case, place-font-style, etc. This just isn't how we control formatting of components within `names`.

Reply:
I think I need to clarify that I meant by "attributes".

I do NOT mean on the names element; I mean on some sub-element, such that it can be used to hold other variables in the future; like (don't take this as a specific recommendation; it's just an illustration):

``` xml 
<name-part variable="place"/>
```

Reply:
Just that `name-part` currently doesn't render variables. It only controls the formatting of a name part. Rendering itself is controlled via `@form` on `cs:name`. 

But I get your point, `place` is not really the same thing as a label since it's not a static string or so, but provided in the data.

What we could do:
- control the rendering and placement of institutional name parts with a new attribute on `cs:name`; `@institutional-form` or so. (Or, we could just allow a list of values as options for `@form`; with some values applying to personal names, others to publishers, e.g. `name form="long institutional-without-place"`.)
- specifiy the delimiter via `@institutional-name-part-delimiter`
- use `name-part name="place"` to control the styling.

Or:
Completely revise the current design to a model where `name-part` actually does control whether a `name-part` is rendered,-i.e. get rid of form here.

Reply:
I really don't think we should deviate from the existing general structure and behavior of `names`. 

The operating behavior of `names` is that it renders the name variable, with the name parts rendered controlled by `@form` and the usage of et-al controlled by `@et-al-min`. The `name-parts` and `et-al` elements only control formatting, not which elements are rendered. 

In contrast, `label` only appears if it is explicitly called. The relative position of `label` and `name` controls the layout of the two pars. That is the key difference between name-parts/et-al and label—the element controls its appearance and its layout. In my view, `label` comes just as much from the data as name-parts do—it's supplied by the variable type (editor, performer, etc.). This is even more clear if we allow user-entered labels to override the defaults (https://github.com/citation-style-language/schema/pull/337). 

So, we have two types "name parts"--those who always appear (the `name-parts` that are children of `name`: family, given, dropping-particle, non-dropping-particle, suffix, institution; `et-al`), and those whose appearance is controlled by the presence of an element (`label`, `abbreviation`, `alternative`, and proposed `place`). 

Following the precedent of `label` getting its own element (and the apparently non-controversial previous additions of `abbreviation` and `alternative`), I suggest we stay consistent and use an element for `place`. I don't really see the advantage of having `<supplemental-name-component="place"/>` or `<supplemental-name-part="abbreviation"/>` versus
`<place/>` or `<abbreviation/>`. That makes styles less readable and is less consistent with `<label/>` being its own element already (which I don't think we should change). 

Additionally, looking at it more closely again, `<alternative/>` does need to be an element because it can take all of the children of `name`. So, if we went with the `<supplemental-name-component="place"/>` structure, then we would have two name components with their own elements (label and alternative) and two name components that are called as attributes on an element (abbreviation and place). I would much prefer to be consistent and use the same structure we have used for `label` consistently.

If we did take the common element approach, we would need a new element other than `name-part`, as that is already used for the specific task of controlling formatting (but not appearance) of name-parts (family, given, suffix, particles, institution). I suggested `supplemental-name-component`. That's obviously very verbose. We could do something like `supplemental-part`. But again, this seems like unnecessary departure from the existing design of `names`. 







 There is already precent for such an element in `names` formatting--